tests/util/caches/test_responsecache.py

```diff
@@ -0,0 +1,125 @@
+# Copyright 2021 Vector Creations Ltd
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from twisted.internet.task import Clock
+
+from synapse.util import Clock as SynapseClock
+from synapse.util.caches.response_cache import ResponseCache
+
+from tests.unittest import TestCase
+
+# few notes about test naming here:
+# 'wait': denotes tests that have an element of "waiting" before its wrapped result becomes available
+# 'expire': denotes tests that test expiry after assured existence
```

Comment:
Does this make sense to be in a docstring instead?

Reply:
Originally it was meant to apply to the whole file, but since the "whole file" now is one class, i'll just move this and add it to that class, probably make it make more sense too. :+1: